src/overlays/actors/ovl_Bg_Dblue_Elevator/z_bg_dblue_elevator.c

```diff
@@ -5,68 +5,373 @@
  */
 
 #include "z_bg_dblue_elevator.h"
+#include "objects/object_dblue_object/object_dblue_object.h"
 
 #define FLAGS (ACTOR_FLAG_10)
 
 #define THIS ((BgDblueElevator*)thisx)
 
-void BgDblueElevator_Init(Actor* thisx, PlayState* play);
+void BgDblueElevator_SpawnRipplesAndSplashes(BgDblueElevator* this, PlayState* play);
+void BgDblueElevator_CheckWaterBoxInfo(BgDblueElevator*, PlayState*);
+s32 BgDblueElevator_GetWaterFlowFromCeiling(Actor* thisx, PlayState* play);
+s32 BgDblueElevator_GetWaterFlowFromPipes(Actor* thisx, PlayState* play);
+
+void BgDblueElevator_Init(Actor* thisx, PlayState* play2);
 void BgDblueElevator_Destroy(Actor* thisx, PlayState* play);
 void BgDblueElevator_Update(Actor* thisx, PlayState* play);
 void BgDblueElevator_Draw(Actor* thisx, PlayState* play);
 
-#if 0
+void BgDblueElevator_SetupWaitForMove(BgDblueElevator* this);
+void BgDblueElevator_WaitForMove(BgDblueElevator* this, PlayState* play);
+void BgDblueElevator_SetupPause(BgDblueElevator* this);
+void BgDblueElevator_Pause(BgDblueElevator* this, PlayState* play);
+void BgDblueElevator_SetupMove(BgDblueElevator* this);
+void BgDblueElevator_Move(BgDblueElevator* this, PlayState* play);
+
 ActorInit Bg_Dblue_Elevator_InitVars = {
-    /**/ ACTOR_BG_DBLUE_ELEVATOR,
-    /**/ ACTORCAT_BG,
-    /**/ FLAGS,
-    /**/ OBJECT_DBLUE_OBJECT,
-    /**/ sizeof(BgDblueElevator),
-    /**/ BgDblueElevator_Init,
-    /**/ BgDblueElevator_Destroy,
-    /**/ BgDblueElevator_Update,
-    /**/ BgDblueElevator_Draw,
+    ACTOR_BG_DBLUE_ELEVATOR,
+    ACTORCAT_BG,
+    FLAGS,
+    OBJECT_DBLUE_OBJECT,
+    sizeof(BgDblueElevator),
+    (ActorFunc)BgDblueElevator_Init,
+    (ActorFunc)BgDblueElevator_Destroy,
+    (ActorFunc)BgDblueElevator_Update,
+    (ActorFunc)BgDblueElevator_Draw,
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80B929EC[] = {
+static BgDblueElevatorStruct1 D_80B92960[4] = {
+    {
+        0,
+        ((s32(*)(BgDblueElevator*, PlayState*))BgDblueElevator_GetWaterFlowFromCeiling),
+        320.0f,
+        30,
+        1,
+        { 0, 0 },
+        1.0f,
+        0.1f,
+        6.0f,
```

Comment:
```suggestion
        6.0f
```
I think this would look better if the individual structs didn't have a trailing comma.